backend/src/simulator.rs

```diff
@@ -6,9 +6,8 @@ use num_bigint::BigUint;
 use num_complex::Complex64;
 use num_traits::{One, Zero};
 use rand::Rng;
-use rustc_hash::FxHashMap;
+use rustc_hash::{FxHashMap, FxHashSet};
```

Comment:
Maybe you can add a comment why using this HashSet over the one in the std library?

Reply:
@msoeken @swernli I imagine for speed of hash implementation [struct.FxHasher.](https://doc.rust-lang.org/stable/nightly-rustc/rustc_data_structures/fx/struct.FxHasher.html)

Reply:
Yes, that's correct. `FxHashMap` has an optimized hashing algorithm that is much faster for rapid access on certain data types, like the `BigUint` we use as a key here.